src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -210,16 +211,21 @@ public void transform(KeyValue<LongWritable, GenericData.Record> input, Emitter<
 
   @Override
   public void onRunFinish(boolean succeeded, BatchSourceContext context) {
-    org.apache.hadoop.fs.Path gcsPath = new org.apache.hadoop.fs.Path(String.format("gs://%s", uuid.toString()));
+    org.apache.hadoop.fs.Path gcsPath = null;
+    String bucket = config.getBucket();
+    if (bucket == null) {
+      gcsPath = new Path(String.format("gs://%s/%s", uuid, uuid));
+    } else {
+      gcsPath = new Path(String.format("gs://%s/%s", bucket, uuid));
+    }
     try {
-      if (config.getBucket() == null) {
-        FileSystem fs = gcsPath.getFileSystem(configuration);
-        if (fs.exists(gcsPath)) {
-          fs.delete(gcsPath, true);
-        }
+      FileSystem fs = gcsPath.getFileSystem(configuration);
+      if (fs.exists(gcsPath)) {
+        fs.delete(gcsPath, true);
```

Comment:
does this also delete temp bucket that was created?

Reply:
Nope, but that is already enabled via AbstractBigQuerySink.java:262. We only delete the bucket if the temporary bucket was created.

Reply:
ah yea - through `fs.gs.bucket.delete.enable`